api/v1beta1/azuremanagedcontrolplane_webhook.go

```diff
@@ -369,13 +369,15 @@ func (m *AzureManagedControlPlane) validateLoadBalancerProfile(_ client.Client)
 func (m *AzureManagedControlPlane) validateAPIServerAccessProfile(_ client.Client) error {
 	if m.Spec.APIServerAccessProfile != nil {
 		var allErrs field.ErrorList
-		for _, ipRange := range m.Spec.APIServerAccessProfile.AuthorizedIPRanges {
-			if _, _, err := net.ParseCIDR(ipRange); err != nil {
-				allErrs = append(allErrs, field.Invalid(field.NewPath("Spec", "APIServerAccessProfile", "AuthorizedIPRanges"), ipRange, "invalid CIDR format"))
+		if m.Spec.APIServerAccessProfile.AuthorizedIPRanges != nil {
```

Comment:
this is redundant, the whole for loop will already get skipped if the array is nil

https://play.golang.com/p/xycugWOnMXp

Reply:
True that. Updating.

Reply:
Updated.